eth/vm/state.py

```diff
@@ -189,6 +204,9 @@ def commit(self, snapshot: Tuple[bytes, Tuple[UUID, UUID]]) -> None:
         _, checkpoint_id = snapshot
         self.account_db.commit(checkpoint_id)
 
+    def persist(self) -> None:
+        self.account_db.persist()
```

Comment:
Would it be accurate to say that nobody should be directly accessing the `state.account_db` (and thus calling methods on it?).  Maybe we should move it to `self._acccount_db`.  If there's a valid use case for pulling the db off the state object, we can add a `get_account_db()` so that people have to choose between:

- `state.persist()`
- `state._account_db.persist()`
- `state.get_account_db().persist()`

And the later two will stand out where-as `state.account_db.persist()`  would just blend in.

Reply:
Yup, I think that could be good. It will make the PR significantly bigger, because of all the current calls to `state.account_db.get_balance()`, etc. But I'll give it a shot.